backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/tournament/service/TournamentTest.groovy

```diff
@@ -0,0 +1,122 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.tournament.service
+
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.BeanConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.SpockTest
+import pt.ulisboa.tecnico.socialsoftware.tutor.config.DateHandler
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.domain.Course
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.domain.CourseExecution
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Assessment
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Question
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Topic
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.TopicConjunction
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.OptionDto
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.TopicDto
+import pt.ulisboa.tecnico.socialsoftware.tutor.tournament.dto.TournamentDto
+import pt.ulisboa.tecnico.socialsoftware.tutor.user.User
+
+import java.time.LocalDateTime
+
+@DataJpaTest
+class TournamentTest extends SpockTest {
+    public static final String STRING_DATE_TODAY = DateHandler.toISOString(DateHandler.now())
+
+    def assessment
+    def topic1
+    def topic2
+    def topics = new HashSet<Integer>()
+    def user1
+
+    def setup() {
+        user1 = createUser(USER_1_NAME, USER_1_USERNAME, USER_1_EMAIL, User.Role.STUDENT, externalCourseExecution)
+
+        def topicDto1 = new TopicDto()
+        topicDto1.setName(TOPIC_1_NAME)
+        topic1 = new Topic(externalCourse, topicDto1)
+        topicRepository.save(topic1)
+
+        def topicDto2 = new TopicDto()
+        topicDto2.setName(TOPIC_2_NAME)
+        topic2 = new Topic(externalCourse, topicDto2)
+        topicRepository.save(topic2)
+
+        topics.add(topic1.getId())
+        topics.add(topic2.getId())
+    }
+
+    def createUser(String name, String username, String email, User.Role role, CourseExecution courseExecution) {
+        def user = new User(name, username, email, role, false, false)
+        user.addCourse(courseExecution)
+        userRepository.save(user)
+        user.setKey(user.getId())
+
+        return user
+    }
+
```

Comment:
redefine these metods using the domain and repositories

Reply:
Done